tls/s2n_client_hello.c

```diff
@@ -298,21 +299,16 @@ int s2n_process_client_hello(struct s2n_connection *conn)
     /* Find potential certificate matches before we choose the cipher. */
     GUARD(s2n_conn_find_name_matching_certs(conn));
 
-    /* Now choose the ciphers and the cert chain. */
-    /* In TLS 1.3, only cipher suite is chosen, and cert chain selection deferred till signature scheme selection */
-    GUARD(s2n_set_cipher_and_cert_as_tls_server(conn, client_hello->cipher_suites.data, client_hello->cipher_suites.size / 2));
+    /* Now choose the ciphers we have certs for. */
+    GUARD(s2n_set_cipher_as_tls_server(conn, client_hello->cipher_suites.data, client_hello->cipher_suites.size / 2));
 
     /* And set the signature and hash algorithm used for key exchange signatures */
     GUARD(s2n_choose_sig_scheme_from_peer_preference_list(conn,
         &conn->handshake_params.client_sig_hash_algs,
         &conn->secure.conn_sig_scheme));
 
-    /* In TLS1.3, certs are not handled by cipher suites, so we must set them here. */
-    if (conn->actual_protocol_version >= S2N_TLS13 ) {
-        s2n_authentication_method auth_method;
-        GUARD(s2n_get_auth_method_from_sig_alg(conn->secure.conn_sig_scheme.sig_alg, &auth_method));
-        conn->handshake_params.our_chain_and_key = s2n_conn_get_compatible_cert_chain_and_key(conn, auth_method);
-    }
+    /* And finally, set the certs specified by the final auth + sig_alg combo. */
+    GUARD(s2n_select_certs_for_server_auth(conn, &conn->handshake_params.our_chain_and_key));
```

Comment:
I know this has been merged already, but I had a question to check my understanding since this bit of code is tricky: Does this change the behavior for a 1.2 server that we had before? This might also involve other recent/related PRs. I'm just bringing it up now because we've made a lot of small changes to all of these functions and you probably have the most recent context on how it all works. I'm thinking of the code we had before we started making changes a couple of months ago (https://github.com/awslabs/s2n/commit/987534357d2e75faf965fca78455eb03734cef6d). 

In the following TLS 1.2 handshake example:
- server has an ECC cert configured
- client sends cipher suite list with only ecdhe_ecdsa_with_aes_128_cbc_sha
- client sends a sig algs list with only rsa_sha1

In the previous behavior, I thought the server would still send it's ECC cert and attempt to stumble forward with the handshake with the default ecdsa_sha1. Would that still happen with this code? Or, would we fail the handshake because we couldn't find a compatible signature scheme from what the client offered?